src/renderer/components/hooks/__tests__/useCheckboxes.spec.js

```diff
@@ -0,0 +1,199 @@
+        synced: true,
+      },
+      {
+        name: 'namespaces-3',
+        synced: true,
+      },
+    ],
+  };
+
+  const testEmptyNamespaces = {
+    namespaces: [],
+  };
+
+  describe('makeCheckboxesInitialState()', () => {
+    it('return initial state for |synced| namespaces', () => {
+      expect(
+        useCheckboxes.makeCheckboxesInitialState(testSyncedNamespaces).parent
+      ).toBe(true);
+      expect(
+        Object.values(
+          useCheckboxes.makeCheckboxesInitialState(testSyncedNamespaces)
+            .children
+        ).every((el) => el === true)
+      ).toBe(true);
+    });
+
+    it('return initial state for |unsynced| namespaces', () => {
+      expect(
+        useCheckboxes.makeCheckboxesInitialState(testUnsyncedNamespaces).parent
+      ).toBe(false);
+      expect(
+        Object.values(
+          useCheckboxes.makeCheckboxesInitialState(testUnsyncedNamespaces)
+            .children
+        ).every((el) => el === false)
+      ).toBe(true);
+    });
+
+    it('return initial state for |mixed| (synced and unsynced) namespaces', () => {
+      expect(
+        useCheckboxes.makeCheckboxesInitialState(testMixedSyncedNamespaces)
+          .parent
+      ).toBe(true);
+      expect(
+        Object.values(
+          useCheckboxes.makeCheckboxesInitialState(testMixedSyncedNamespaces)
+            .children
+        ).some((el) => el === true) &&
+          Object.values(
+            useCheckboxes.makeCheckboxesInitialState(testMixedSyncedNamespaces)
+              .children
+          ).some((el) => el === false)
+      ).toBe(true);
+    });
+  });
+
+  describe('useCheckboxes()', () => {
+    describe('getSyncedData()', () => {
+      it('get synced and ignored namespaces', () => {
+        const TestComponent = ({ testData }) => {
+          const { getSyncedData } = useCheckboxes.useCheckboxes(
+            useCheckboxes.makeCheckboxesInitialState(testData)
+          );
+          const { syncedNamespaces, ignoredNamespaces } = getSyncedData();
+
+          return (
+            <div>
+              <p>{syncedNamespaces}</p>
+              <p>{ignoredNamespaces}</p>
+            </div>
+          );
+        };
+
+        render(<TestComponent testData={testMixedSyncedNamespaces} />);
+
+        const ignoredNamespaces = [];
+        const syncedNamespaces = [];
+
+        Object.keys(
+          useCheckboxes.makeCheckboxesInitialState(testMixedSyncedNamespaces)
+            .children
+        ).forEach((name) => {
+          if (
+            useCheckboxes.makeCheckboxesInitialState(testMixedSyncedNamespaces)
+              .children[name]
+          ) {
+            syncedNamespaces.push(name);
+          } else {
+            ignoredNamespaces.push(name);
+          }
+        });
+
+        expect(screen.getByText(syncedNamespaces.join(''))).toBeInTheDocument();
+        expect(
+          screen.getByText(ignoredNamespaces.join(''))
+        ).toBeInTheDocument();
```

Comment:
I think these expectations would be stronger if you were more explicit about which text you were comparing to.

In your `TestComponent`, I would do this:
```javascript
          return (
            <div>
              <p data-testid="synced">{syncedNamespaces}</p>
              <p data-testid="ignored">{ignoredNamespaces}</p>
            </div>
          );
```

And here, I would do this:
```suggestion
        expect(screen.getByTestId('synced')).toHaveTextContent(syncedNamespaces.join(''));
        expect(
          screen.getByTestId('ignored')).toHaveTextContent(ignoredNamespaces.join(''));
```